src/Microsoft.Azure.WebJobs.Host/Lease/LeaseFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Lease;
+using Microsoft.Azure.WebJobs.Host.Storage;
+
+namespace Microsoft.Azure.WebJobs.Host.Lease
+{
+    /// <summary>
+    /// ILeaseProxy factory
+    /// </summary>
+    internal class LeaseFactory
+    {
+        public static ILeaseProxy CreateLeaseProxy(IStorageAccountProvider storageAccountProvider)
+        {
+            ILeaseProxy leaseProxy = null;
+
+            // Choose between Sql and Blob based lease implementations
+            if (SqlLeaseProxy.IsSqlLeaseType())
```

Comment:
Agree with @jvano, this would hurt functions startup time. We may still want to test the sql connection after we've established that it looks like a sql connection string, however, in order to flow exceptions to the runtime / fall back to storage if necessary.